### PR TITLE
refactor `GetJobHandler` data layer

### PR DESCRIPTION
**Pull request by justinpjose**
### What
**Describe what you have changed and why.**
- Component test added for empty job id given when getting a job
- Return pointer to models.Job and return nil when there is an error (for `GetJob` and `FindJob`)

### How to review
**Describe the steps required to test the changes.**
- Check if the changes make sense
- Check if all the test passes

### Who can review
**Describe who worked on the changes, so that other people can review.**
- Anyone

### CODE REVIEW

File: api/jobs.go [hunk 1]
```diff
@@ -128,14 +128,16 @@ func (api *API) CreateJobHandler(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
-// GetJobHandler returns a function that gets an existing Job resource, from the Job Store, that's associated with the id passed in.
+// GetJobHandler returns a function that gets an existing Job resource, from the Job Store, that's associated with the id passed in
 func (api *API) GetJobHandler(w http.ResponseWriter, req *http.Request) {
 	ctx := req.Context()
 	host := req.Host
+
```
Comment (rahulPalliyalil) [inline api/jobs.go:135]: (suggestion): extra space ?

File: api/jobs.go [hunk 2]
```diff
@@ -155,29 +158,23 @@ func (api *API) GetJobHandler(w http.ResponseWriter, req *http.Request) {
...
-		http.Error(w, serverErrorMessage, http.StatusInternalServerError)
-		return
-	}
-	w.Header().Set("Content-Type", "application/json")
-	_, err = w.Write(jsonResponse)
-	if err != nil {
-		logData["json_response"] = jsonResponse
-		log.Error(ctx, "writing response failed", err, logData)
+		logData["response_status_to_write"] = http.StatusOK
+
```
Comment (rahulPalliyalil) [inline api/jobs.go:170]: (suggestion): extra space ?

File: features/v1_features/latest_version_features/getting_a_job.feature [hunk 3]
```diff
@@ -20,6 +20,17 @@ Feature: Getting a job
       | total_search_documents          | 0                         |
       | total_inserted_search_documents | 0                         |
 
+  Scenario: When request is made with empty job id and request returns StatusNotFound by gorilla/mux as handler is not found
+
+    Given the number of existing jobs in the Job Store is 0
+    And the api version is undefined for incoming requests
+    When I GET "/jobs/"
```
Comment (rahulPalliyalil) [inline features/v1_features/latest_version_features/getting_a_job.feature:27]: (minor): you only need ```/jobs``` here ?

File: features/v1_features/latest_version_features/getting_a_job.feature [hunk 4]
```diff
@@ -20,6 +20,17 @@ Feature: Getting a job
       | total_search_documents          | 0                         |
       | total_inserted_search_documents | 0                         |
 
+  Scenario: When request is made with empty job id and request returns StatusNotFound by gorilla/mux as handler is not found
```
Comment (rahulPalliyalil) [inline features/v1_features/latest_version_features/getting_a_job.feature:23]: (minor): I guess you probably don't need to specify the router library (gorilla/mux) here as this component test should run against any router library and should not be dependent on a specific library.

**Review by rahulPalliyalil — CHANGES_REQUESTED** [review]

File: api/jobs.go (hunk 1 above)
Comment (justinpjose) [reply api/jobs.go:126]: Done

File: api/jobs.go (hunk 2 above)
Comment (justinpjose) [reply api/jobs.go:161]: Done

File: features/v1_features/latest_version_features/getting_a_job.feature (hunk 3 above)
Comment (justinpjose) [reply features/v1_features/latest_version_features/getting_a_job.feature:27]: Done

File: features/v1_features/latest_version_features/getting_a_job.feature (hunk 4 above)
Comment (justinpjose) [reply features/v1_features/latest_version_features/getting_a_job.feature:23]: Done

**Review by rahulPalliyalil — APPROVED** [review]

**Pull request merged**